packages/holocron/src/externalRegistry.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Map as iMap, fromJS } from 'immutable';
+import semver from 'semver';
+
+let registeredExternals = iMap();
+let requiredModuleExternals = iMap();
+
+function getRegisteredExternals() {
+  return registeredExternals.toJS();
+}
+
+function validateExternal({ providedVersion, requestedRange }) {
+  return semver.satisfies(providedVersion, requestedRange);
+}
+
+function registerExternal({ name, version, module }) {
```

Comment:
using the term `module` here is a bit confusing. My first thought was it is a one-app module, but it looks like this is the actual code for the dependency. Maybe consider changing this to `code` ?
```suggestion
function registerExternal({ name, version, code }) {
```

Reply:
I can see that, they are npm modules.

Reply:
Maybe we just add a JSDOC above this function to make it clear?